src/patternfly/components/TreeView/tree-view.scss

```diff
@@ -0,0 +1,196 @@
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-2__list-item__link--PaddingLeft);
+      }
+    }
+  }
+
+  &.pf-m-nested-level-3 {
+    .pf-c-tree-view__list-item {
+      &.pf-m-expandable > .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-3__list-item--m-expandable__link--PaddingLeft);
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-3__list-item__link--PaddingLeft);
+      }
+    }
+  }
+}
+// stylelint-enable
+
+.pf-c-tree-view__list-item {
+  &.pf-m-expanded {
+    > .pf-c-tree-view__link > .pf-c-tree-view__link-toggle {
+      transform: rotate(var(--pf-c-tree-view__link--m-expanded__toggle--Rotate));
+    }
+  }
+}
+
+.pf-c-tree-view__link {
+  display: flex;
+  align-items: center;
+  width: 100%;
+  padding: var(--pf-c-tree-view__link--PaddingTop) var(--pf-c-tree-view__link--PaddingRight) var(--pf-c-tree-view__link--PaddingBottom) var(--pf-c-tree-view__link--PaddingLeft);
+  color: var(--pf-c-tree-view__link--Color);
+  text-align: left;
+  border: 0;
+
+  &.pf-m-current {
+    --pf-c-tree-view__link--Color: var(--pf-c-tree-view__link--m-current--Color);
+
+    font-weight: var(--pf-c-tree-view__link--m-current--FontWeight);
+  }
+
+  &.pf-m-first {
+    font-weight: var(--pf-c-tree-view__link--m-first--FontWeight);
+  }
+
+  &:hover {
+    background-color: var(--pf-c-tree-view__link--hover--BackgroundColor);
+  }
+
+  .pf-c-check {
+    --pf-c-check__input--MarginTop: 0;
+    --pf-c-check--GridGap: 0;
+
+    padding-right: var(--pf-c-tree-view__link--c-check--PaddingRight);
+  }
+
+  .pf-c-badge {
+    margin-left: var(--pf-c-tree-view__link--c-badge--MarginLeft);
+
+    &.pf-m-read {
+      --pf-c-badge--m-read--BackgroundColor: var(--pf-c-tree-view__link--c-badge--BackgroundColor);
+    }
+  }
+}
+
+.pf-c-tree-view__link-toggle {
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  min-width: var(--pf-c-tree-view__link-toggle--MinWidth);
+  min-height: var(--pf-c-tree-view__link-toggle--MinHeight);
+  padding-right: var(--pf-c-tree-view__link-toggle--PaddingRight);
+  padding-left: var(--pf-c-tree-view__link-toggle--PaddingLeft);
+}
+
+.pf-c-tree-view__link-text {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+
+.pf-c-tree-view__search {
+  padding: var(--pf-c-tree-view__search--PaddingTop) var(--pf-c-tree-view__search--PaddingRight) var(--pf-c-tree-view__search--PaddingBottom) var(--pf-c-tree-view__search--PaddingLeft);
+}
+
+.pf-c-tree-view__link-icon {
+  padding-right: var(--pf-c-tree-view__link-icon--PaddingRight);
+  color: var(--pf-c-tree-view__link-icon--Color);
+}
+
+.pf-c-tree-view__link-wrapper {
+  display: grid;
+  grid-template-columns: minmax(0, 100%) auto;
+
+  &:hover {
```

Comment:
I'm confused about the lack of ability to tab/focus to things. I would expect the background change to happen on focus, and with this item, `:focus-within`. Also `.pf-c-tree-view__action` is only visible on `:hover` - what about keyboard users?

Reply:
Is this still a concern? or does what i said about managing focus via JS cover it?

Reply:
Are you suggesting I add focus-within to the wrapper (now called node) element ?